cvmfs/util/algorithm.h

```diff
@@ -107,6 +106,56 @@ class StopWatch : SingleCopy {
   timeval start_, end_;
 };
 
+/**
+ * Log2Histogram is a simple implementation of 
+ * log2 histogram data structure which stores 
+ * and prints log2 histogram. It is used for
+ * getting and printing latency metrics of 
+ * CVMFS fuse calls.
+ * 
+ * Log2Histogram hist(2);
+ * hist.Add(1);
+ * hist.Add(2);
+ * hist.PrintLog2Histogram();
+ */
+
+class Log2Histogram {
+ public:
+  explicit Log2Histogram(unsigned int nbins);
+
+  void Add(unsigned int value) {
+    unsigned int i;
+    unsigned int flag = 1;
+
+    for (i = 1; i <= this->bins_.size(); i++) {
+      if (value < this->boundary_values_[i]) {
+        atomic_inc32(&(this->bins_[i]));
+        flag = 0;
+        return;
+      }
+    }
+    if (flag) {
+      atomic_inc32(&(this->bins_[0]));  // add to overflow bin.
+    }
+  }
+
+  std::string ToString();
+
+  void PrintLog2Histogram();
+
+  friend class UTLog2Histogram;
+
+ private:
+  std::vector<atomic_int32> bins_;
+  // boundary_values_ handle the largest value a certain
+  // bin can store in itself.
+  std::vector<unsigned int> boundary_values_;
+};
+
+class UTLog2Histogram {
```

Comment:
Add a comment that this class is a helper for the unit tests to extract internals from `Log2Histogram`.  You can even move the class to the t_util.cc file.